types/openid-client/index.d.ts

```diff
@@ -89,6 +90,22 @@ export interface EndSessionUrlParameters {
     readonly state?: string;
 }
 
+export interface IntrospectionResponse {
+    active: boolean;
+    scope?: string[];
+    client_id?: string;
+    username?: string;
+    token_type?: string;
+    exp?: number;
+    iat?: number;
+    nbf?: number;
+    sub?: string;
+    aud?: string;
+    iss?: string;
+    jti?: string;
+    [key: string]: any;
```

Comment:
I would prefer `{} | null | undefined` instead of `any` for the indexer.

This forces the callers to think about the result and do an explicit cast (except for the props you added above).

Reply:
... also please add `readonly` modifiers to all members (favor immutability)

Reply:
`any` is probably a bit too forgiving here. 

How would you feel about using `unknown` instead? It would still require explicit casting, and would be more indicative of the intent behind the response.

Reply:
Ah ... we're already at TS 3.0 ... yes, use `unknown`